### PR TITLE
VMware: New module vmware_host_sriov

### PR DESCRIPTION
**Pull request by victron**
##### SUMMARY
Add a new module for changing SR-IOV settings on host interface

##### ISSUE TYPE
- New Module Pull Request

##### COMPONENT NAME
vmware_host_sriov

##### ADDITIONAL INFORMATION
- This module can be used to configure, enable/disable SR-IOV functions an ESXi host.
- module didn't reboot host after changes, but put in output "rebootRequired" key from vCenter status
- tested on VMware 6.0

### CODE REVIEW

File: lib/ansible/modules/cloud/vmware/vmware_host_sriov.py [hunk 1]
```diff
@@ -0,0 +1,332 @@
...
+
+DOCUMENTATION = r"""
+---
+module: vmware_host_sriov
+short_description: Manage SR-IOV settings on host
+description:
+- This module can be used to configure, enable/disable SR-IOV functions an ESXi host.
+- module didn't reboot host after changes, but put in output "rebootRequired" key.
+- User can specify an ESXi hostname or Cluster name. In case of cluster name, all ESXi hosts are updated.
+version_added: 2.9
```
Comment (Akasurde) [inline lib/ansible/modules/cloud/vmware/vmware_host_sriov.py]:
```suggestion
version_added: 2.10
```

File: lib/ansible/modules/cloud/vmware/vmware_host_sriov.py [hunk 2]
```diff
@@ -0,0 +1,332 @@
...
+author:
+- Viktor Tsymbalyuk (@victron)
+notes:
+- Tested on vSphere 6.0
+requirements:
+- python >= 2.6
```
Comment (Akasurde) [inline lib/ansible/modules/cloud/vmware/vmware_host_sriov.py]:
```suggestion
- python >= 2.7
```

Pyvmomi requires 2.7

Comment (Akasurde) [reply lib/ansible/modules/cloud/vmware/vmware_host_sriov.py]: I know I need to change for the rest of them.

File: lib/ansible/modules/cloud/vmware/vmware_host_sriov.py [hunk 3]
```diff
@@ -0,0 +1,332 @@
...
+    type: str
+  cluster_name:
+    description:
+    - Name of the cluster from which all host systems will be used.
+    - This parameter is required if C(esxi_hostname) is not specified.
+    type: str
+  vmnic:
+    description:
+    - interface name, like vmnic0
+    type: str
```
Comment (Akasurde) [inline lib/ansible/modules/cloud/vmware/vmware_host_sriov.py:46]: Can this be a list of NICs ?

File: lib/ansible/modules/cloud/vmware/vmware_host_sriov.py [hunk 4]
```diff
@@ -0,0 +1,332 @@
...
+    required: True
+  sriovOn:
```
Comment (Akasurde) [inline lib/ansible/modules/cloud/vmware/vmware_host_sriov.py]:
```suggestion
  sriov_on:
```

File: lib/ansible/modules/cloud/vmware/vmware_host_sriov.py [hunk 5]
```diff
@@ -0,0 +1,332 @@
...
+    description:
+    - desired SR-IOV state on interface
```
Comment (Akasurde) [inline lib/ansible/modules/cloud/vmware/vmware_host_sriov.py]:
```suggestion
    - Desired SR-IOV state on interface.
```

File: lib/ansible/modules/cloud/vmware/vmware_host_sriov.py [hunk 6]
```diff
@@ -0,0 +1,332 @@
...
+    type: bool
+    required: True
+  numVirtFunc:
```
Comment (Akasurde) [inline lib/ansible/modules/cloud/vmware/vmware_host_sriov.py]:
```suggestion
  num_virt_func:
```

File: lib/ansible/modules/cloud/vmware/vmware_host_sriov.py [hunk 7]
```diff
@@ -0,0 +1,332 @@
...
+    description:
+    - number of functions to activate on interface
```
Comment (Akasurde) [inline lib/ansible/modules/cloud/vmware/vmware_host_sriov.py]:
```suggestion
    - number of functions to activate on interface.
```

File: lib/ansible/modules/cloud/vmware/vmware_host_sriov.py [hunk 8]
```diff
@@ -0,0 +1,332 @@
...
+        self.vmnic = self.params.get("vmnic", None)
+        self.sriovOn = self.params.get("sriovOn", None)
+        self.numVirtFunc = self.params.get("numVirtFunc", None)
+
+        self.hosts = self.get_all_host_objs(
+            cluster_name=cluster_name, esxi_host_name=esxi_host_name
+        )
+        if not self.hosts:
+            self.module.fail_json(msg="Failed to find host system.")
+        # self.change_list = []
```
Comment (Akasurde) [inline lib/ansible/modules/cloud/vmware/vmware_host_sriov.py]:
```suggestion

```

**Review by Akasurde — CHANGES_REQUESTED** [review]
Overall looks good. @goneri  @pgbidkar @anusha94 @Tomorrow9 Can you please review this ? Thanks.

File: lib/ansible/modules/cloud/vmware/vmware_host_sriov.py (hunk 3 above)
Comment (victron) [reply lib/ansible/modules/cloud/vmware/vmware_host_sriov.py:46]: I thought about it. But in reason that it's my first experience, I prefer leave it as is.  A little bit late, I'll redo it on [str,  list].

File: lib/ansible/modules/cloud/vmware/vmware_host_sriov.py [hunk 9]
```diff
@@ -0,0 +1,331 @@
...
+    "status": ["preview"],
+    "supported_by": "community",
+}
+
+DOCUMENTATION = r"""
+---
+module: vmware_host_sriov
+short_description: Manage SR-IOV settings on host
+description:
+- This module can be used to configure, enable/disable SR-IOV functions an ESXi host.
```
Comment (pgbidkar) [inline lib/ansible/modules/cloud/vmware/vmware_host_sriov.py]:
```suggestion
- This module can be used to configure, enable/disable SR-IOV functions on ESXi host.
```

File: lib/ansible/modules/cloud/vmware/vmware_host_sriov.py [hunk 10]
```diff
@@ -0,0 +1,331 @@
...
+- module didn't reboot host after changes, but put in output "rebootRequired" key.
```
Comment (pgbidkar) [inline lib/ansible/modules/cloud/vmware/vmware_host_sriov.py]: Please can you rephrase this sentence? Something lile
```suggestion
- Module doesn't  reboot the host after changes, but puts it in output "rebootRequired" state.
```

File: lib/ansible/modules/cloud/vmware/vmware_host_sriov.py [hunk 11]
```diff
@@ -0,0 +1,331 @@
...
+notes:
+- Tested on vSphere 6.0
+requirements:
+- python >= 2.7
+- PyVmomi
+options:
+  esxi_hostname:
+    description:
+    - Name of the host system to work with.
+    - This parameter is required if C(cluster_name) is not specified.
```
Comment (pgbidkar) [inline lib/ansible/modules/cloud/vmware/vmware_host_sriov.py:35]: I think user can specify specific host from cluster. Cluster and host both arguments should be allowed

File: lib/ansible/modules/cloud/vmware/vmware_host_sriov.py [hunk 12]
```diff
@@ -0,0 +1,331 @@
...
+    type: str
+  cluster_name:
+    description:
+    - Name of the cluster from which all host systems will be used.
+    - This parameter is required if C(esxi_hostname) is not specified.
```
Comment (pgbidkar) [inline lib/ansible/modules/cloud/vmware/vmware_host_sriov.py:41]: I think user can specify specific host from cluster. Cluster and host both arguments should be allowed

File: lib/ansible/modules/cloud/vmware/vmware_host_sriov.py [hunk 13]
```diff
@@ -0,0 +1,331 @@
...
+  vmnic:
+    description:
+    - interface name, like vmnic0
+    type: str
+    required: True
+  sriov_on:
+    description:
+    - Desired SR-IOV state on interface.
+    type: bool
+    required: True
```
Comment (pgbidkar) [inline lib/ansible/modules/cloud/vmware/vmware_host_sriov.py]: is default value not required here?

File: lib/ansible/modules/cloud/vmware/vmware_host_sriov.py [hunk 14]
```diff
@@ -0,0 +1,331 @@
...
+        )
+        if not self.hosts:
+            self.module.fail_json(msg="Failed to find host system.")
+        self.results = {"before": {}, "after": {}, "changes": {}}
+
+    def set_host_state(self):
+        """Check ESXi host configuration"""
+        change_list = []
+        changed = False
+        for host in self.hosts:
```
Comment (pgbidkar) [inline lib/ansible/modules/cloud/vmware/vmware_host_sriov.py:270]: is this loop applicable only Cluster name is passed right?

File: lib/ansible/modules/cloud/vmware/vmware_host_sriov.py [hunk 15]
```diff
@@ -0,0 +1,331 @@
...
+            )
+            return False
+
+
+def main():
+    """Main"""
+    argument_spec = vmware_argument_spec()
+    argument_spec.update(
+        cluster_name=dict(type="str", required=False),
+        esxi_hostname=dict(type="str", required=False),
```
Comment (pgbidkar) [inline lib/ansible/modules/cloud/vmware/vmware_host_sriov.py:364]: cluster_name or esxi_hostname any one of them should be required.

File: lib/ansible/modules/cloud/vmware/vmware_host_sriov.py [hunk 16]
```diff
@@ -0,0 +1,331 @@
...
+        vmnic=dict(type="str", required=True),
+        sriov_on=dict(type="bool", required=True),
+        num_virt_func=dict(type="int", required=True),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        required_one_of=[["cluster_name", "esxi_hostname"], ["sriov_on", "num_virt_func"]],
+        supports_check_mode=True,
```
Comment (pgbidkar) [inline lib/ansible/modules/cloud/vmware/vmware_host_sriov.py:373]: I think this not handled?

File: test/integration/targets/vmware_host_sriov/tasks/main.yml [hunk 17]
```diff
@@ -0,0 +1,102 @@
+# Test code for the  module vmware_host_sriov
+
+- when: vcsim is not defined
```
Comment (pgbidkar) [inline test/integration/targets/vmware_host_sriov/tasks/main.yml:3]: This block will skip all tests in integration test. @goneri please can you confirm if this block is required?

**Review by pgbidkar — CHANGES_REQUESTED** [review]

File: lib/ansible/modules/cloud/vmware/vmware_host_sriov.py [hunk 18]
```diff
@@ -0,0 +1,331 @@
...
+  sriov_on:
+    description:
+    - Desired SR-IOV state on interface.
+    type: bool
+    required: True
+  num_virt_func:
+    description:
+    - number of functions to activate on interface.
+    - if sriov_on false should be equal 0
+    - if sriov_on true should be more then 0
```
Comment (anusha94) [inline lib/ansible/modules/cloud/vmware/vmware_host_sriov.py]: if sriov_on true should be more than 0

**Review by anusha94 — COMMENTED** [review]
Since there are scenarios like 
`either one of esxi_host or cluster is required`, 
`if sriov_on is false, then num_virt_func = 0`, these can be checked for in the beginning itself and you can `exit_json` with appropriate messages

IMO, it is better to have a function to sanitize params like in [vmware_guest_network](https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/cloud/vmware/vmware_guest_network.py#L331)

File: lib/ansible/modules/cloud/vmware/vmware_host_sriov.py (hunk 13 above)
Comment (victron) [reply lib/ansible/modules/cloud/vmware/vmware_host_sriov.py]: as I understood it done especially:
`./ansible/lib/ansible/module_utils/common/parameters.py`
```
    for (k, v) in argument_spec.items():
        legal_inputs.append(k)
        aliases = v.get('aliases', None)
        default = v.get('default', None)
        required = v.get('required', False)
        if default is not None and required:
            # not alias specific but this is a good place to check this
            raise ValueError("internal error: required and default are mutually exclusive for %s" % k)
        if aliases is None:
```
 I got this error
```
{"failed": true, "msg": "Module alias error: internal error: required and default are mutually exclusive for num_virt_func"}
```

File: lib/ansible/modules/cloud/vmware/vmware_host_sriov.py (hunk 14 above)
Comment (victron) [reply lib/ansible/modules/cloud/vmware/vmware_host_sriov.py:270]: I got this error
```
{"failed": true, "msg": "Module alias error: internal error: required and default are mutually exclusive for num_virt_func"}
```

Comment (pgbidkar) [reply lib/ansible/modules/cloud/vmware/vmware_host_sriov.py:270]: I didnt get this. Did you make any changes to loop and got this error?

**Comment by victron** [comment]
> 
> 
> Since there are scenarios like
> `either one of esxi_host or cluster is required`,
> `if sriov_on is false, then num_virt_func = 0`, these can be checked for in the beginning itself and you can `exit_json` with appropriate messages
> 
> IMO, it is better to have a function to sanitize params like in [vmware_guest_network](https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/cloud/vmware/vmware_guest_network.py#L331)

thanks @anusha94 , thank you for advice I'll do it.

File: test/integration/targets/vmware_host_sriov/tasks/main.yml (hunk 17 above)
Comment (goneri) [reply test/integration/targets/vmware_host_sriov/tasks/main.yml:3]: The question is more: Can we run the integration test-suite with govcsim? if it's the case, we should drop the line. Otherwise, the integration tests will still be run, but only with a regular lab.

**Comment by victron** [comment]
main changes in coomit
```
num_virt_func:
    description:
    - number of functions to activate on interface.
    - 0 means SR-IOV disabled.
    - number greater than 0 means SR-IOV enabled.
    type: int
    required: True
  sriov_on:
    description:
    - optional parameter, related to num_virt_func.
    - SR-IOV can be enabled only if num_virt_func > 0.
    type: bool
    required: False
```
and added  function to sanitize params.
Please make a review.

File: lib/ansible/modules/cloud/vmware/vmware_host_sriov.py (hunk 3 above)
Comment (victron) [reply lib/ansible/modules/cloud/vmware/vmware_host_sriov.py:46]: I'll do it in next version. Because may be it's better use dict instead, for example:
```
[{ "vmnic": "vmnic0",  "num_virt_func": 8},
{ "vmnic": "vmnic1",  "num_virt_func": 0}]

File: lib/ansible/modules/cloud/vmware/vmware_host_sriov.py (hunk 11 above)
Comment (victron) [reply lib/ansible/modules/cloud/vmware/vmware_host_sriov.py:35]: yes, it allowed.  I added this comment in doc.
`- user can specify specific host from cluster.`

File: lib/ansible/modules/cloud/vmware/vmware_host_sriov.py (hunk 12 above)
Comment (victron) [reply lib/ansible/modules/cloud/vmware/vmware_host_sriov.py:41]: yes, it allowed. I added this comment in doc.
`- user can specify specific host from cluster.`

File: lib/ansible/modules/cloud/vmware/vmware_host_sriov.py (hunk 14 above)
Comment (victron) [reply lib/ansible/modules/cloud/vmware/vmware_host_sriov.py:270]: sorry, ignore my previous post - wrong thread.
> is this loop applicable only Cluster name is passed right?

yes, you are right

File: lib/ansible/modules/cloud/vmware/vmware_host_sriov.py (hunk 15 above)
Comment (victron) [reply lib/ansible/modules/cloud/vmware/vmware_host_sriov.py:364]: yes, but not both of them in a same time.
this check done in
```
module = AnsibleModule(
        argument_spec=argument_spec,
        required_one_of=[["cluster_name", "esxi_hostname"]],
```
in
`./ansible/lib/ansible/module_utils/common/validation.py`
```
def check_required_one_of(terms, module_parameters):
    """Check each list of terms to ensure at least one exists in the given module
    parameters

    Accepts a list of lists or tuples
```

File: lib/ansible/modules/cloud/vmware/vmware_host_sriov.py (hunk 16 above)
Comment (victron) [reply lib/ansible/modules/cloud/vmware/vmware_host_sriov.py:373]: it, handled. If to look into
`./ansible/lib/ansible/module_utils/common/validation.py`
```
def check_required_one_of(terms, module_parameters):
    """Check each list of terms to ensure at least one exists in the given module
    parameters

    Accepts a list of lists or tuples
```
but now it doesn't matter, I simplified  this part. 
```
    module = AnsibleModule(
        argument_spec=argument_spec,
        required_one_of=[["cluster_name", "esxi_hostname"]],
        supports_check_mode=True,
    )
```

File: lib/ansible/modules/cloud/vmware/vmware_host_sriov.py (hunk 15 above)
Comment (pgbidkar) [reply lib/ansible/modules/cloud/vmware/vmware_host_sriov.py:364]: What I meant is one of them should be made required=True. 
What happens if user doesnt pass both of them?

File: test/integration/targets/vmware_host_sriov/tasks/main.yml (hunk 17 above)
Comment (victron) [reply test/integration/targets/vmware_host_sriov/tasks/main.yml:3]: as I understood there are no possibility to test it with `govcsim`
```
$ curl -sk https://user:pass@127.0.0.1:8989/about | grep -i configManager  
FAIL: 1
```
May be I am wrong,  please point me.  

> if it's the case, we should drop the line. therwise, the integration tests will still be run, but only with a regular lab.

I thought, that `- when: vcsim is not defined` - instructs not to use `vcsim` and forward directly to regular lab tests. It's not correct?

File: lib/ansible/modules/cloud/vmware/vmware_host_sriov.py (hunk 15 above)
Comment (victron) [reply lib/ansible/modules/cloud/vmware/vmware_host_sriov.py:364]: > What happens if user doesnt pass both of them?

all will be fine
```
vic@AN: ~/ansible vmware_host_sriov
$ cat ./t_args.json                                                                                      [14:25:08]
{
        "ANSIBLE_MODULE_ARGS": {
                "hostname": "10.72.24.143",
                "username": "administrator@vsphere.local", 
                "password": "Aff", 
                "validate_certs": false, 
                "sriov_on": true,
                "num_virt_func": 8, 
                "vmnic": "vmnic7"
        }
}

(envPy37_ansible_dev)  
vic@AN: ~/ansible vmware_host_sriov
$ python -m ansible.modules.cloud.vmware.vmware_host_sriov ./t_args.json                                 [14:25:30]

{"msg": "one of the following is required: cluster_name, esxi_hostname", "failed": true, "invocation": {"module_args": {"hostname": "10.72.24.143", "username": "administrator@vsphere.local", "password": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER", "validate_certs": false, "sriov_on": true, "num_virt_func": 8, "vmnic": "vmnic7", "port": 443}}}
(envPy37_ansible_dev) FAIL: 1

vic@AN: ~/ansible vmware_host_sriov
```

File: lib/ansible/modules/cloud/vmware/vmware_host_sriov.py [hunk 19]
```diff
@@ -0,0 +1,387 @@
...
+    description:
+    - number of functions to activate on interface.
+    - 0 means SR-IOV disabled.
+    - number greater than 0 means SR-IOV enabled.
+    type: int
+    required: True
+  sriov_on:
+    description:
+    - optional parameter, related to num_virt_func.
+    - SR-IOV can be enabled only if num_virt_func > 0.
```
Comment (anusha94) [inline lib/ansible/modules/cloud/vmware/vmware_host_sriov.py]: SR-IOV can be enabled only if `C(num_virt_func) > 0.`

File: lib/ansible/modules/cloud/vmware/vmware_host_sriov.py [hunk 20]
```diff
@@ -0,0 +1,387 @@
...
+  num_virt_func:
+    description:
+    - number of functions to activate on interface.
+    - 0 means SR-IOV disabled.
+    - number greater than 0 means SR-IOV enabled.
+    type: int
+    required: True
+  sriov_on:
+    description:
+    - optional parameter, related to num_virt_func.
```
Comment (anusha94) [inline lib/ansible/modules/cloud/vmware/vmware_host_sriov.py]: optional parameter, related to `C(num_virt_func)`.

File: lib/ansible/modules/cloud/vmware/vmware_host_sriov.py [hunk 21]
```diff
@@ -0,0 +1,387 @@
...
+
+        #  prepare list of hosts to work with them
+        self.hosts = self.get_all_host_objs(
+            cluster_name=cluster_name, esxi_host_name=esxi_host_name
+        )
+        if not self.hosts:
+            self.module.fail_json(msg="Failed to find host system.")
+        self.results = {"before": {}, "after": {}, "changes": {}}
+
+    def sanitize_params(self, before, hostname, vmnic):
```
Comment (anusha94) [inline lib/ansible/modules/cloud/vmware/vmware_host_sriov.py]: I am not clear as to why `before` `hostname` and `vmnic` are required, whatever the input parameters can be obtained from `module.params`

File: lib/ansible/modules/cloud/vmware/vmware_host_sriov.py [hunk 22]
```diff
@@ -0,0 +1,387 @@
...
+            if self.sriov_on is False:
+                self.module.fail_json(
+                    msg="with sriov_on == false,  alowed value for num_virt_func is 0"
+                )
+            params_to_change.update(
+                {"sriovEnabled": True, "numVirtualFunction": self.num_virt_func}
+            )
+
+            # check compatability
+            if not before["sriovCapable"]:
```
Comment (anusha94) [inline lib/ansible/modules/cloud/vmware/vmware_host_sriov.py:216]: checking compatibility can be a different method. It doesn't have to be under `sanitize_params`.
Or change the method name to something more appropriate

File: lib/ansible/modules/cloud/vmware/vmware_host_sriov.py [hunk 23]
```diff
@@ -0,0 +1,387 @@
...
+        for host in self.hosts:
+            self.results["before"][host.name] = {}
+            self.results["after"][host.name] = {}
+            self.results["changes"][host.name] = {}
+            self.results["before"][host.name] = self._check_sriov(host)
+            self.results["changes"][host.name]["rebootRequired"] = self.results[
+                "before"
+            ][host.name]["rebootRequired"]
+
+            params_to_change = self.sanitize_params(
```
Comment (anusha94) [inline lib/ansible/modules/cloud/vmware/vmware_host_sriov.py]: If you choose to make `sanitize_params` and `check_compatibility` separate methods, `sanitize_params` can be called even before the `for loop` above

File: lib/ansible/modules/cloud/vmware/vmware_host_sriov.py [hunk 24]
```diff
@@ -0,0 +1,387 @@
...
+            if (
+                before["numVirtualFunctionRequested"]
+                != params_to_change["numVirtualFunction"]
+            ):
+                params_to_change["changes"]["numVirtualFunction"] = params_to_change[
+                    "numVirtualFunction"
+                ]
+                params_to_change["change"] = True
+            else:
+                params_to_change["changes"]["msg"] = "Not active (looks not rebooted) "
```
Comment (anusha94) [inline lib/ansible/modules/cloud/vmware/vmware_host_sriov.py]: Not active (looks **like** not rebooted)?

**Review by anusha94 — CHANGES_REQUESTED** [review]

File: lib/ansible/modules/cloud/vmware/vmware_host_sriov.py (hunk 19 above)
Comment (victron) [reply lib/ansible/modules/cloud/vmware/vmware_host_sriov.py]: ok, thanks

File: lib/ansible/modules/cloud/vmware/vmware_host_sriov.py (hunk 20 above)
Comment (victron) [reply lib/ansible/modules/cloud/vmware/vmware_host_sriov.py]: ok, thanks

File: lib/ansible/modules/cloud/vmware/vmware_host_sriov.py (hunk 21 above)
Comment (victron) [reply lib/ansible/modules/cloud/vmware/vmware_host_sriov.py]: ok, now I got your point. I'll redo.

File: lib/ansible/modules/cloud/vmware/vmware_host_sriov.py (hunk 24 above)
Comment (victron) [reply lib/ansible/modules/cloud/vmware/vmware_host_sriov.py]: ok

File: lib/ansible/modules/cloud/vmware/vmware_host_sriov.py (hunk 22 above)
Comment (victron) [reply lib/ansible/modules/cloud/vmware/vmware_host_sriov.py:216]: thank you, I'll do separate method.

File: lib/ansible/modules/cloud/vmware/vmware_host_sriov.py (hunk 23 above)
Comment (victron) [reply lib/ansible/modules/cloud/vmware/vmware_host_sriov.py]: thanks, it will look more clear.

File: lib/ansible/modules/cloud/vmware/vmware_host_sriov.py [hunk 25]
```diff
@@ -184,117 +184,112 @@ def __init__(self, module):
...
         if self.num_virt_func < 0:
             self.module.fail_json(msg="allowed value for num_virt_func >= 0")
-
         if self.num_virt_func == 0:
             if self.sriov_on is True:
                 self.module.fail_json(
                     msg="with sriov_on == true,  alowed value for num_virt_func > 0"
                 )
-            params_to_change.update({"sriovEnabled": False, "numVirtualFunction": 0})
+            self.sriov_on = False  # fill value, if urser not provided
```
Comment (anusha94) [inline lib/ansible/modules/cloud/vmware/vmware_host_sriov.py]: typo for `user`

Comment (anusha94) [reply lib/ansible/modules/cloud/vmware/vmware_host_sriov.py]: instead of setting to `False` here, can you make it as a default value if not provided?

File: lib/ansible/modules/cloud/vmware/vmware_host_sriov.py [hunk 26]
```diff
@@ -184,117 +184,112 @@ def __init__(self, module):
...
 
         if self.num_virt_func > 0:
             if self.sriov_on is False:
                 self.module.fail_json(
                     msg="with sriov_on == false,  alowed value for num_virt_func is 0"
                 )
-            params_to_change.update(
-                {"sriovEnabled": True, "numVirtualFunction": self.num_virt_func}
-            )
+            self.sriov_on = True  # fill value, if urser not provided
```
Comment (anusha94) [inline lib/ansible/modules/cloud/vmware/vmware_host_sriov.py]: same as above, since `sriov_on` is an optional arg, looks like it can be handled with assigning a default value.

**Review by anusha94 — COMMENTED** [review]
Just a few minor observations. Please look into it.

File: lib/ansible/modules/cloud/vmware/vmware_host_sriov.py (hunk 25 above)
Comment (victron) [reply lib/ansible/modules/cloud/vmware/vmware_host_sriov.py]: I think not. Because I need give two values into `nic_sriov = vim.host.SriovConfig()`.
PyVmomi (VMware) requires two parameters. So, at this point it's better to set correct parameter.

Comment (anusha94) [reply lib/ansible/modules/cloud/vmware/vmware_host_sriov.py]: okay, seems right.
Just the typo correction for `user` is required.
Everything else looks good :)

Comment (victron) [reply lib/ansible/modules/cloud/vmware/vmware_host_sriov.py]: I did.
Anusha, thank you very much for your advices. Now it looks better )

**Review by anusha94 — APPROVED** [review]
shipit

**Review by goneri — APPROVED** [review]

**Comment by goneri** [comment]
Thanks @victron for the patch and @anusha94 for the review. I don't have the hardware to test the module, but it looks really good to me :-).

**Comment by Akasurde** [comment]
Superseded by https://github.com/ansible-collections/vmware/pull/267
